### ISSUE TITLE
release 0.7.0

### ISSUE THREAD

**Issue by dwalton76** [rhempel/ev3dev-lang-python#207]
@WasabiFan I think you said there is one more tweak to make to the README but it is waiting on a PR for the docs to go through correct?  After that I think we can do a release.

**Comment by WasabiFan** [comment]
Ok, sounds good. I am waiting on a review/merge of https://github.com/ev3dev/ev3dev.github.io/pull/237 so that the README isn't confusing to new users. After that, I think this'll be ready.

**Comment by dwalton76** [comment]
These are the items that are currently in the changelog for 0.7.0
- Drop python 2.x support.
- Performance improvements for reading/writing sysfs attributes.
- Updates for breaking ev3dev kernel changes.

Should we mention that many demos have been added? They are not installed as part of the package but in the changelog we could point people to https://github.com/rhempel/ev3dev-lang-python/tree/develop/demo I think we should add this to the changelog but I am biased as I did a lot of work on the demo programs.  Proposed text:
- Added many demo programs including BALANC3R, MINDCUB3R, EXPLOR3R, TRACK3R and EV3D4. A touchscreen web interface is available for tank robots, see EV3D4WebControl. The demo programs are available at https://github.com/rhempel/ev3dev-lang-python/tree/develop/demo

Other things to add:
- support for ev3dev kernel  4.4.15-13-ev3dev-ev3
- Added ev3dev/helper.py which includes classes for Tanks, Motors, Web UI controls, etc
- Added ev3dev/GyroBalancer.py for robots that use a gyroscope to balance on two wheels

**Comment by dlech** [comment]
> Should we mention that many demos have been added?

Not in the debian changelog. If there is another changelog, then sure, mention the demos if you like.

**Comment by WasabiFan** [comment]
> Should we mention that many demos have been added?

We should definitely advertise them, whether it is in the changelog or somewhere else. If it's in something version-specific, we should make sure to link to the version of the tree at the release tag instead of `master` to make sure it works in the future.

----

Also note that the important kernel version number is the `13`; the rest of it is specific to the platform. So, we should say "`-13-ev3dev` kernels".

**Comment by WasabiFan** [comment]
The dependency that we had on ev3dev.org content has fallen into place, so I have opened #209 to update the README accordingly. Once that is merged we should be ready to go.

**Comment by dwalton76** [comment]
On a related note....anyone know why the demo programs never shipped as part of the package?  If we want people to try them out or reference them it would be much easier if they were installed via the package to somewhere like /usr/share/ev3dev-lang-python/

**Comment by WasabiFan** [comment]
I think it's because they are not essential to the library core, so one generally doesn't want it to install unless explicitly requested (especially now that there is JavaScript in that folder as well). You're right, though; it would be nice to get them into people's hands without hassle. Maybe a separate `python-ev3dev-demos` package?

**Comment by WasabiFan** [comment]
Once this new version is merged, I think we should switch the default branch to `master` so that people see the repo as it was at the last release by default. Thoughts?

**Comment by dwalton76** [comment]
That's kinda what /usr/share/ is for though (docs, examples, etc)...but we are close enough to doing a release we can figure out exactly how to get the demos out there after this.

**Comment by dwalton76** [comment]
I think we should keep it at develop because that is what we want folks to make pull requests against.  If we switch it to master someone will dev something against it unintentionally and will then have to rebase to develop and go through merge conflicts.

**Comment by WasabiFan** [comment]
> I think we should keep it at develop because that is what we want folks to make pull requests against.

OK, that makes sense. We need to make sure that people don't look at new content and get confused, though. I, for one, often open up a GitHub repo to look up line numbers in error messages, and when repos have their default branch set to something other than the most recent release it can get confusing. Let's leave it like this and see how it goes.

**Comment by dlech** [comment]
https://www.debian.org/doc/debian-policy/ch-docs.html#s12.6

**Comment by WasabiFan** [comment]
I've drafted a release for v0.7.0 [here](https://github.com/rhempel/ev3dev-lang-python/releases/tag/untagged-7ecb33c66536d0b2174f) (only viewable by project collaborators). I included the content of that release description below as well. Other than #221, is there anything else that we need to do before releasing? We should start to think about when we want to align with semver and call a release 1.0.

To perform the actual release, we'll need to merge `develop` into `master` (and use the `--no-ff` option to force a merge commit, or do it through a PR) and then hit the release button to create the tag. Once we have created the release tag, someone will need to go through the process of publishing the Debian package from that tag.

**Comment by ddemidov** [comment]
May be add a link to the kernel announcement where the changed to the motor API were made?

**Comment by dwalton76** [comment]
I get a 404 on the change log link. Other than that it looks good to me.

**Comment by WasabiFan** [comment]
Yeah, the link is to the changelog on the nonexistent 0.7.0 tag. Once we publish the release, that link should start working.

@ddemidov David is going to include a more explanatory section on ev3dev API changes in the release announcement for the upcoming release, so I think I'll edit in a link to that after it is posted.

**Comment by WasabiFan** [comment]
Do we have any volunteers to do the Debian packaging stuff after we release? I am prepared to release this library within the next 12 hours, assuming I get others who OK it.

**Comment by WasabiFan** [comment]
Ok... New plan :wink: **I will release this library in 12 hours unless someone says otherwise**. It is currently 7:20 in my time zone (-8:00), so I'll tag the release around 19:00. Someone else will have to do the debian stuff though.

Also, I think I am going to change the procedure I mentioned earlier: I am not going to force a merge commit. Assuming git is willing, I'll do a fast-forward.

**Comment by ddemidov** [comment]
I have no objections re version tagging. 

Who took care of the Debian packaging last time? Are there some kind of instructions? Is it possible to do without having a Debian system, or just with ev3?

**Comment by WasabiFan** [comment]
I can't grab the link now, but there's a page on the site entitled "packaging for ev3dev" which has instructions. It doesn't require a straight Debian system, but I believe you do need a derivative (Ubuntu, Elementary, etc.).

**Comment by ddemidov** [comment]
Found it: http://www.ev3dev.org/docs/devtools/packaging-for-ev3dev/
OK, I'll try to do this after you tag 0.7.0.

**Comment by dwalton76** [comment]
Awesome thank you @WasabiFan   Sorry I haven't been much help this week I am travelling for work.

**Comment by WasabiFan** [comment]
> Sorry I haven't been much help this week I am travelling for work.

No problem! The amount of work required here is pretty small, it's just a matter of making sure that I don't surprise anyone with a release.

**Comment by WasabiFan** [comment]
✨ BAM! New release. https://github.com/rhempel/ev3dev-lang-python/releases/tag/0.7.0

**Comment by WasabiFan** [comment]
I find it interesting that it already says I released it 11 hours ago 😕

**Comment by WasabiFan** [comment]
@ddemidov The Read the Docs site has updated to the new documentation content (I think) but the version number is still the old one. Where is that configured?

**Comment by ddemidov** [comment]
It gets the version from the last annotated tag with git describe through
git_version.py. But the tag has to be annotated (created with git tag - a).

**Comment by ddemidov** [comment]
I've retagged the same commit with annotated tag and forced pushed it to github. Hope that's ok.

**Comment by WasabiFan** [comment]
The release still seems happy, so I think it's fine. Thanks!

**Comment by ddemidov** [comment]
Do we update the [debian copyright file](https://github.com/rhempel/ev3dev-lang-python/blob/develop/debian/copyright#L7-L10) with all the contributors names/emails? I we do, I guess I'll need an ok from @dwalton (`Daniel Walton <dwalton76@gmail.com>`), @fryfrog (`Donals Webster, <fryfrog@gmail.com>`), @251 (`Frank Busse <bb0xfb@gmail.com>`), @WasabiFan (`Kaelin Laundry <wasabifan@outlook.com>`) to include their names and emails into the copyright file.

**Comment by ddemidov** [comment]
> Do we update the debian copyright file with all the contributors names/emails? 

Opened #224 for this

**Comment by dlech** [comment]
> Do we update the debian copyright file with all the contributors names/emails?

If they claim copyright on their work. They could also assign copyright to someone else if they choose.

**Comment by ddemidov** [comment]
I've just uploaded the debian and raspbian packages for 0.7.0

**Comment by ddemidov** [comment]
```
sudo apt-get update
sudo apt-get install python3-ev3dev
```
should update the library to version 0.7.0

**Comment by WasabiFan** [comment]
Thanks for doing this! I'll update the release description to remove the warning about this. Can we close this issue now?

**Comment by dlech** [comment]
package is included in https://oss.jfrog.org/list/oss-snapshot-local/org/ev3dev/brickstrap/2016-09-30/
